### PR TITLE
8923 op mode

### PR DESCRIPTION
**Pull request by daaniaabbas**
Created base Opmodes including TeleOp and autonomous. Not tested yet.

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (malavika1) [inline .idea/.gitignore:1]: I think the repo should already have a .gitignore file so you shouldn't need to add this.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseOpMode.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        motorFL.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        motorFR.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        motorBL.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+        motorBR.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
+
+        motorFL.setDirection(DcMotor.Direction.FORWARD);
+        motorFR.setDirection(DcMotor.Direction.REVERSE);
+        motorBL.setDirection(DcMotor.Direction.REVERSE);
+        motorBR.setDirection(DcMotor.Direction.REVERSE);
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseOpMode.java:40]: I feel like two should be forward and two should be reverse, but I also don't really remember how it works for mecanum drive.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseTeleOp.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+abstract public class BaseTeleOp extends BaseOpMode {
+
+    private boolean isSlowMode = false;
+
+    private Toggle driveSpeedToggle = new Toggle();
+
+    double driveSpeed = 1.0;
+
+    public void driveRobot(){
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseTeleOp.java:11]: https://google.github.io/styleguide/javaguide.html#s4.1-braces
```suggestion
    public void driveRobot() {
```

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseTeleOp.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        double y = gamepad1.left_stick_y;
+        double x = gamepad1.left_stick_x;
+        double rotationalPower = gamepad1.right_stick_x;
+    }
+
+    public void driveRobotSpeed(){
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BaseTeleOp.java:17]:
```suggestion
    public void driveRobotSpeed() {
```

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BlueLeftDetectPark.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package org.firstinspires.ftc.team8923_PowerPlay;
+
+public class BlueLeftDetectPark {
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BlueLeftDetectPark.java:3]: Are Left and Right going to be clear to you in the game? Maybe you can use different terminology based on the field.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package org.firstinspires.ftc.team8923_PowerPlay;
+
+public class Constants {
+    //ratios
+    static final double TICKS_PER_REVOLUTION = 537.6; // Neverest orbital 20, 7 pulse per revolution
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:5]: Do you know what type of motor you're using?

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    static final double GEAR_RATIO = 1.0;
+    static final double WHEEL_DIAMETER = 2.5; // inches
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:7]: Make sure to update the gear ratio and the wheel diameter when you know what they are. You can leave a TODO comment.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    public static final double ROTATION_I = 0.0;
+    public static final double ROTATION_D = 0.0;
+    public static final double TRANSLATION_P = 0.0004;
+    public static final double TRANSLATION_I = 0.0;
+    public static final double TRANSLATION_D = 0.0;
+    public static final double ANGLE_TOLERANCE_DEG = 5.0;
+    public static final double POSITION_TOLERANCE_MM = 2*25.4;
+
+    //intake
+    public static final double INTAKE_PWR = 1;
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:28]: You can delete this for now and add in mechanism code later.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Toggle.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    public Toggle() {
+
+    }
+
+    public boolean toggle(boolean button)
+    {
+        if(press)
+        {
+            toggleState = !toggleState;
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Toggle.java:16]:
```suggestion
        if (press) {
            toggleState = !toggleState;
```

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Toggle.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        if(! prevState && button)
+        {
+            press = true;
+        }
+        else if(press)
+        {
+            press = false;
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Toggle.java:25]:
```suggestion
        if (!prevState && button) {
            press = true;
        } else if (press) {
            press = false;
```

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BlueLeftDetectPark.java (hunk 5 above)
Comment (daaniaabbas) [reply Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/BlueLeftDetectPark.java:3]: Yes, after looking at the field we can determine which side to start the robot on based on the alliance we're on. If we're blue alliance, we would stand on the blue alliance side and be either left or right and the same for the red alliance. There shouldn't be confusion between the left and right once we decide on our alliance. I hope that makes sense.

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java (hunk 6 above)
Comment (daaniaabbas) [reply Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:5]: They should be the same motors from last year. I'll double check when I'm in the shop

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java (hunk 7 above)
Comment (Sanj8923) [reply Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:7]: Wheel diameter is 4 inches

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java (hunk 6 above)
Comment (Sanj8923) [reply Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:5]: they are orbital 20s

File: Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package org.firstinspires.ftc.team8923_PowerPlay;
+
+public class Constants {
+    //ratios
+    static final double TICKS_PER_REVOLUTION = 537.6; // Neverest orbital 20, 7 pulse per revolution
+    static final double GEAR_RATIO = 1.0;
+    static final double WHEEL_DIAMETER = 4; // inches
```
Comment (malavika1) [inline Team8923_PowerPlay/src/main/java/org/firstinspires/ftc/team8923_PowerPlay/Constants.java:7]:
```suggestion
    static final double WHEEL_DIAMETER = 4.0; // inches
```

**Review by malavika1 — APPROVED** [review]

**Pull request merged**